src/SimpleGraphs/simplegraph.jl

```diff
@@ -23,7 +23,7 @@ function SimpleGraph(
     return SimpleGraph{T}(ne, fadjlist)
 end
 
-eltype(x::SimpleGraph{T}) where T = T
+eltype(::Type{SimpleGraph{T}}) where T = T
```

Comment:
```suggestion
eltype(::Type{<: AbstractSimpleGraph{T}}) where T = T
```

should also work in my opinion. But then we should move that definition to `src/SimpleGraphs/SimpleGraphs.jl` and also remove the similar one for `SimpleDiGraph`

Reply:
Moved it to AbstractGraph instead